src/backend/distributed/commands/truncate.c

```diff
@@ -363,113 +359,46 @@ ExecuteTruncateStmtSequentialIfNecessary(TruncateStmt *command)
 static void
 LockTruncatedRelationMetadataInWorkers(TruncateStmt *truncateStatement)
 {
-	List *distributedRelationList = NIL;
-
 	/* nothing to do if there is no metadata at worker nodes */
 	if (!ClusterHasKnownMetadataWorkers())
 	{
 		return;
 	}
 
+	List *distributedRelationList = NIL;
+	List *referencingRelationIds = NIL;
+
 	RangeVar *rangeVar = NULL;
 	foreach_ptr(rangeVar, truncateStatement->relations)
 	{
 		Oid relationId = RangeVarGetRelid(rangeVar, NoLock, false);
-		Oid referencingRelationId = InvalidOid;
 
 		if (!IsCitusTable(relationId))
 		{
 			continue;
 		}
 
-		if (list_member_oid(distributedRelationList, relationId))
-		{
-			continue;
-		}
-
-		distributedRelationList = lappend_oid(distributedRelationList, relationId);
+		distributedRelationList = list_append_unique_oid(distributedRelationList,
+														 relationId);
 
 		CitusTableCacheEntry *cacheEntry = GetCitusTableCacheEntry(relationId);
 		Assert(cacheEntry != NULL);
 
 		List *referencingTableList = cacheEntry->referencingRelationsViaForeignKey;
+
+		Oid referencingRelationId = InvalidOid;
 		foreach_oid(referencingRelationId, referencingTableList)
 		{
-			distributedRelationList = list_append_unique_oid(distributedRelationList,
-															 referencingRelationId);
+			referencingRelationIds = lappend_oid(referencingRelationIds,
```

Comment:
why do we maintain two lists (e.g., distributedRelationList and referencingRelationIds).

Can't we have only distributedRelationList? I'm fine always calling `list_append_unique_oid` in the loop as well

Reply:
Will add a comment to clarify why two lists are used.